src/exporters/vhi.py

```diff
@@ -0,0 +1,220 @@
+        """  get the filenames containing VHI """
+        with FTP('ftp.star.nesdis.noaa.gov') as ftp:
+            ftp.login()
+            ftp.cwd('/pub/corp/scsb/wguo/data/Blended_VH_4km/VH/')
+
+            # append the filenames to a list
+            listing: List = []
+            ftp.retrlines("LIST", listing.append)
+            # extract the filename
+            filepaths = [f.split(' ')[-1] for f in listing]
+            # extract only the filenames of interest
+            vhi_files = [f for f in filepaths if ".VH.nc" in f]
+
+        return vhi_files
+
+    # @staticmethod
+    def chunks(self, l: List, n: int) -> Generator:
+        # return a generator object which chunks list into sublists of size n
+        # https://chrisalbon.com/python/data_wrangling/break_list_into_chunks_of_equal_size/
+        # For item i in a range that is a length of l,
+        for i in range(0, len(l), n):
+            # Create an index range for l of n items:
+            yield l[i:i + n]
+
+    def save_errors(self, outputs: List) -> None:
+        print("\nError: ", [errors for errors in outputs if errors is not None])
+
+        # save the filenames that failed to a pickle object
+        with open(self.raw_folder / 'vhi_export_errors.pkl', 'wb') as f:
+            pickle.dump([error[-1] for error in outputs if error is not None], f)
+
+        return
+
+    def run_parallel(self,
+                     vhi_files: List,
+                     years: List[int] = [yr for yr in np.arange(1981, 2020)]
+                     ) -> List:
+        pool = Pool(processes=100)
+
+        # split the filenames into batches of 100
+        batches = [batch for batch in self.chunks(vhi_files, 100)]
+
+        # run in parallel for multiple file downloads
+        # passing mutliple args using partial https://stackoverflow.com/a/5442981/9940782
+        args = dict(raw_folder=self.raw_folder, years=years)
+        outputs = pool.map(partial(batch_ftp_request, args), batches)
+
+        # write the output (TODO: turn into logging behaviour)
+        print("\n\n*************************")
+        print("VHI Data Downloaded")
+        print("*************************")
+        print("Errors:")
+        pprint([error for error in outputs if error is not None])
+        # save errors
+        self.save_errors(outputs)
+
+        return batches
+
+    def check_failures(self) -> List:
+        """ Read the outputted list of errors to the user """
+        pickled_error_fname = "vhi_export_errors.pkl"
+        assert (self.raw_folder / pickled_error_fname).exists(), f"the file:\
+         {(self.raw_folder / 'vhi_export_errors.pkl')} \
+         does not exist! Required to check the files that failed"
+
+        with open(pickled_error_fname, 'rb') as f:
+            errors = pickle.load(f)
+        print("*************************")
+        print("Errors:")
+        pprint(errors)
+
+        return errors
+
+    def export(self) -> List:
```

Comment:
Could we add the following to this function:

- The option to change the arguments of the download (right now, I can't limit the number of years I download by calling `export`)
- Add a docstring, since this is the function users will use to interact with the exporter

Reply:
Also, in your notebook you have a nice assert statement which makes sure the years passed are within the accepted range. It would make sense for that to be within this export function too

Reply:
which notebook?

Reply:
okay fixed